### ISSUE TITLE
connect to snowflake returns generic FDW error.

### ISSUE THREAD

**Issue by andre-aura** [supabase/wrappers#386]
# Bug report

- [X] I confirm this is a bug with Supabase, not with my own application. 
- [X] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Hi Folks!

I followed the steps described [here](https://supabase.com/docs/guides/database/extensions/wrappers/snowflake) to create the wrapper and the external tables.

When querying Snowflake I got this error:
```
ERROR:  HV000: guest fdw error: expected value at line 1 column 1
```
There is nor more details at postgres logs.

I'm using the same key that I use to connect via dbt.

I can't see any query against snowflake with my user.

Also, I tried to create the same structure `mytable` from the docs and still no sucess.

What I'm missing here?

Thanks!

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to '…'
2. Click on '…'
3. Scroll down to '…'

## System information

- OS: [e.g. macOS, Windows]
- Browser (if applies) [e.g. chrome, safari]
- Version of supabase-js: [e.g. 6.0.2]
- Version of Node.js: [e.g. 10.10.0]

**Comment by burmecia** [comment]
Thanks for reporting this issue. Could you provide more details such as Postgres version, Wrappers version and DDL used to create tables on both Snowflake and Postgres side?

**Comment by jonathansergio** [comment]
Hey @burmecia and @andre-aura! The expected value at line 1 column 1 is a classic serde_json parsing error. It looks like when Snowflake rejects the connection (e.g., auth failure), it returns a non-JSON body (like an HTML error page or plain text). The wrapper is blindly attempting to deserialize this error response into JSON, which panics serde.

I can open a quick PR to inspect the HTTP status code first. If it's a 4xx/5xx, we can read the raw text response and bubble it up to the user via a clean HV000 Postgres error, making it much easier to debug their Snowflake credentials. Does that sound good?

**Comment by burmecia** [comment]
@jonathansergio yes that's the cause and really appreciate it if you can make the error reporting better!

**Comment by sanskar-soni-9** [comment]
Picking this up unless there is any WIP @jonathansergio  @burmecia
